Sources/Views/MessageLabel.swift

```diff
@@ -22,417 +22,491 @@
+            }
+            
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
-
-    open override var font: UIFont! {
+    
+    public var font: UIFont = UIFont.systemFont(ofSize: 10.0) {
         didSet {
-            guard font != oldValue else { return }
-            guard let attributedText = attributedText else { return }
-            textStorage.setAttributedString(attributedText)
-            setNeedsDisplay()
+            textStorage.addAttribute(key: .font, value: font)
+            updateDetectorAttributes()
+            if !isConfiguring {
+                setNeedsDisplay()
+            }
         }
     }
-
-    open override var textColor: UIColor! {
+    
+    public var textColor: UIColor = .darkText {
         didSet {
-            guard textColor != oldValue else { return }
-            guard let attributedText = attributedText else { return }
-            textStorage.setAttributedString(attributedText)
-            setNeedsDisplay()
+            textStorage.addAttribute(key: .foregroundColor, value: textColor)
+            updateDetectorAttributes()
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
-
-    open override var lineBreakMode: NSLineBreakMode {
+    
+    public var lineBreakMode: NSLineBreakMode = .byWordWrapping {
         didSet {
-            guard lineBreakMode != oldValue else { return }
             textContainer.lineBreakMode = lineBreakMode
-            setNeedsDisplay()
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
-
-    open override var numberOfLines: Int {
+    
+    public var lineFragmentPadding: CGFloat = 0 {
+        didSet {
+            textContainer.lineFragmentPadding = lineFragmentPadding
+            if !isConfiguring { setNeedsDisplay() }
+        }
+    }
+    
+    public var numberOfLines: Int = 0 {
         didSet {
-            guard numberOfLines != oldValue else { return }
             textContainer.maximumNumberOfLines = numberOfLines
-            setNeedsDisplay()
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
-
-    open override var textAlignment: NSTextAlignment {
+    public var textAlignment: NSTextAlignment = .left {
         didSet {
-            guard textAlignment != oldValue else { return }
-            setTextStorage()
+            let style = paragraphStyle(from: textStorage.attributedString)
+            textStorage.addAttribute(key: .paragraphStyle, value: style)
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
 
-    open var textInsets: UIEdgeInsets = .zero {
+    public var textInsets: UIEdgeInsets = .zero {
         didSet {
-            guard textInsets != oldValue else { return }
-            setNeedsDisplay()
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
-
-    open var addressAttributes: [NSAttributedStringKey: Any] = [:] {
+    
+    public var enabledDetectors: [DetectorType] = [.address, .phoneNumber, .date, .url] {
```

Comment:
From a performance standpoint wouldn't it be better to have the default detectors be empty? Experienced users can then set the detectors they want

Reply:
Yeah, we definitely will default to none. These will be overriden by the `TextMessageDisplayDelegate` that defaults to `[]` anyways but I will clean it up

Reply:
👍